# Домашнее задание 
# Тема: Действия с мультииндексами в датафреймах

In [1]:
import pandas as pd
import numpy as np

## Задание 1
### Постройте датафрейм заполненный нулями так, как это показано в ячейке ниже. Первые две колонки являются мультииндексом. Для заполнения датафрейма нулями воспользуйтесь функцией zeros из библиотеки numpy

In [6]:
index = [['Группа 1', 'Группа 1', 'Группа 2', 'Группа 2'],
         ['Альфа','Бета','Альфа','Бета']]
columns = [['Владивосток', 'Владивосток', 'Казань', 'Новосибирск', 'Новосибирск'],
         ['Кол-во','Сумма','Сумма','Кол-во','Сумма']]
zer = (len(index[1]),len(columns[1]))
data = np.zeros(zer)

In [8]:
df1_result = pd.DataFrame(data, index=index, columns=columns)
df1_result

Владивосток       Казань Новосибирск      
                    Кол-во Сумма  Сумма      Кол-во Сумма
Группа 1 Альфа         0.0   0.0    0.0         0.0   0.0
         Бета          0.0   0.0    0.0         0.0   0.0
Группа 2 Альфа         0.0   0.0    0.0         0.0   0.0
         Бета          0.0   0.0    0.0         0.0   0.0

### Получившийся датафрейм сохраните в csv файл и загрузите на степик

In [9]:
df1_result.to_csv('df1_result.csv')

## Задание 2
### Загрузите датафрейм из файла task5.csv. Поменяйте местами уровни так чтобы сначала шел тип парковки (платная/бесплатная), затем округ и район. Отсортируйте так чтобы сначала шли бесплатные парковки в алфавитном порядке по округу, а внутри округа в алфавитном порядке по району*. 

#### *Некоторые названия районов записаны наоборот: вместо район Нижегородский записано Нижегородский район. Из-за этого сортировка по алфавиту будет идти не совсем так, как вы ожидали. Оставим результат сортировки по алфавиту как есть (переносить слово район для корректной сортировки не будем).

In [16]:
df2_task = pd.read_csv('task5.csv', index_col=[0, 1, 2], header=[0, 1]) # не редактируйте ячейку

In [17]:
df2_task

Capacity  \
                                                                           count   
Восточный административный округ Нижегородский район         Бесплатная        1   
                                 район Богородское           Бесплатная        2   
                                                             Платная           1   
                                 район Вешняки               Бесплатная        5   
                                                             Платная           3   
...                                                                          ...   
Южный административный округ     район Чертаново Северное    Платная           2   
                                 район Чертаново Центральное Бесплатная        7   
                                                             Платная           3   
                                 район Чертаново Южное       Бесплатная        9   
                                                             Платная           4   

                                                                             \
                                                                        sum   
Восточный административный округ Нижегородский район         Бесплатная   6   
                                 район Богородское           Бесплатная  14   
                                                             Платная      8   
                                 район Вешняки               Бесплатная  35   
                                                             Платная     17   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная     20   
                                 район Чертаново Центральное Бесплатная  68   
                                                             Платная     21   
                                 район Чертаново Южное       Бесплатная  86   
                                                             Платная     36   

                                                                        rating  \
                                                                           min   
Восточный административный округ Нижегородский район         Бесплатная      8   
                                 район Богородское           Бесплатная      8   
                                                             Платная         0   
                                 район Вешняки               Бесплатная      2   
                                                             Платная         0   
...                                                                        ...   
Южный административный округ     район Чертаново Северное    Платная         2   
                                 район Чертаново Центральное Бесплатная      0   
                                                             Платная         3   
                                 район Чертаново Южное       Бесплатная      3   
                                                             Платная         0   

                                                                             \
                                                                        max   
Восточный административный округ Нижегородский район         Бесплатная   8   
                                 район Богородское           Бесплатная  10   
                                                             Платная      0   
                                 район Вешняки               Бесплатная  10   
                                                             Платная      9   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная      6   
                                 район Чертаново Центральное Бесплатная   9   
                                                             Платная      9   
               

In [18]:
df2_task = df2_task.swaplevel(1, 2)
df2_task = df2_task.swaplevel(0, 1)

In [19]:
df2_task

Capacity  \
                                                                           count   
Бесплатная Восточный административный округ Нижегородский район                1   
                                            район Богородское                  2   
Платная    Восточный административный округ район Богородское                  1   
Бесплатная Восточный административный округ район Вешняки                      5   
Платная    Восточный административный округ район Вешняки                      3   
...                                                                          ...   
           Южный административный округ     район Чертаново Северное           2   
Бесплатная Южный административный округ     район Чертаново Центральное        7   
Платная    Южный административный округ     район Чертаново Центральное        3   
Бесплатная Южный административный округ     район Чертаново Южное              9   
Платная    Южный административный округ     район Чертаново Южное              4   

                                                                             \
                                                                        sum   
Бесплатная Восточный административный округ Нижегородский район           6   
                                            район Богородское            14   
Платная    Восточный административный округ район Богородское             8   
Бесплатная Восточный административный округ район Вешняки                35   
Платная    Восточный административный округ район Вешняки                17   
...                                                                      ..   
           Южный административный округ     район Чертаново Северное     20   
Бесплатная Южный административный округ     район Чертаново Центральное  68   
Платная    Южный административный округ     район Чертаново Центральное  21   
Бесплатная Южный административный округ     район Чертаново Южное        86   
Платная    Южный административный округ     район Чертаново Южное        36   

                                                                        rating  \
                                                                           min   
Бесплатная Восточный административный округ Нижегородский район              8   
                                            район Богородское                8   
Платная    Восточный административный округ район Богородское                0   
Бесплатная Восточный административный округ район Вешняки                    2   
Платная    Восточный административный округ район Вешняки                    0   
...                                                                        ...   
           Южный административный округ     район Чертаново Северное         2   
Бесплатная Южный административный округ     район Чертаново Центральное      0   
Платная    Южный административный округ     район Чертаново Центральное      3   
Бесплатная Южный административный округ     район Чертаново Южное            3   
Платная    Южный административный округ     район Чертаново Южное            0   

                                                                             \
                                                                        max   
Бесплатная Восточный административный округ Нижегородский район           8   
                                            район Богородское            10   
Платная    Восточный административный округ район Богородское             0   
Бесплатная Восточный административный округ район Вешняки                10   
Платная    Восточный административный округ район Вешняки                 9   
...                                                                      ..   
           Южный административный округ     район Чертаново Северное      6   
Бесплатная Южный административный округ     район Чертаново Центральное   9   
Платная    Южный административный округ     район Чертаново Центральное   9   
Бесплатная Южны

In [25]:
df2_result = df2_task.sort_index(level = [0,1,2])

#### Обратите внимание, наш датафрейм немного усложнился: у мультииндекса строк пропали названия для уровней (добавлять их не стоит) и появился мультииндекс для колонок. Теперь в датафрейме содержится дополнительная информация о рейтинге парковки: маинимальное, максимальное и среднее значение

In [29]:
df2_result = df2_result.head(27)
df2_result

Capacity  \
                                                                          count   
Бесплатная Восточный административный округ Нижегородский район               1   
                                            район Богородское                 2   
                                            район Вешняки                     5   
                                            район Восточное Измайлово         1   
                                            район Гольяново                   2   
                                            район Ивановское                  2   
                                            район Измайлово                   2   
                                            район Косино-Ухтомский            3   
                                            район Метрогородок                4   
                                            район Новогиреево                 2   
                                            район Новокосино                  1   
                                            район Перово                      5   
                                            район Преображенское              3   
                                            район Северное Измайлово          1   
                                            район Соколиная Гора              4   
                                            район Сокольники                  4   
           Западный административный округ  Можайский район                   1   
                                            район Дорогомилово                1   
                                            район Крылатское                  2   
                                            район Кунцево                     1   
                                            район Ново-Переделкино            4   
                                            район Очаково-Матвеевское         1   
                                            район Проспект Вернадского        2   
                                            район Раменки                     4   
                                            район Солнцево                    2   
                                            район Тропарёво-Никулино          1   
                                            район Филёвский Парк              1   

                                                                            \
                                                                       sum   
Бесплатная Восточный административный округ Нижегородский район          6   
                                            район Богородское           14   
                                            район Вешняки               35   
                                            район Восточное Измайлово    8   
                                            район Гольяново              9   
                                            район Ивановское            16   
                                            район Измайлово             11   
                                            район Косино-Ухтомский      18   
                                            район Метрогородок          18   
                                            район Новогиреево           11   
                                            район Новокосино             1   
                                            район Перово                35   
                                            район Преображенское        21   
                                            район Северное Измайлово     1   
                                            район Соколиная Гора        16   
                                            район Сокольники            28   
           Западный административный округ  Можайский район              8   
                                            район Дорогомилово           8   
                                            район Крылатское            18   
                         

### Сохраните на степик только первые 27 записей датафрейма.

In [30]:
df2_result.to_csv('df2_result.csv')

## Задание 3
### Загрузите датафрейм из файла task5.csv. Из мультииндекса строк сбросьте уровень в котором содержится информация о типе парковки. Сохраните данные из этого уровня в отдельную колонку под названием `type`. Из целого датафрейма вырезаем только данные о рейтинге, сортируем записи по среднему рейтингу в порядке убывания и загружаем на степик ТОП-10 записей.

In [74]:
df3_task = pd.read_csv('task5.csv', index_col=[0, 1, 2], header=[0, 1]) # не редактируйте ячейку

In [75]:
df3_task = df3_task['rating'].sort_values(by = 'mean', ascending = False).head(10)
df3_task.reset_index(level = 2, col_level=1 )

level_2  min  \
Юго-Восточный административный округ  район Некрасовка      Бесплатная   10   
Зеленоградский административный округ район Силино             Платная   10   
Северный административный округ       район Сокол              Платная   10   
Западный административный округ       район Марьино            Платная   10   
                                      район Филёвский Парк  Бесплатная   10   
Юго-Восточный административный округ  район Печатники       Бесплатная    9   
Западный административный округ       Можайский район       Бесплатная    9   
                                      район Дорогомилово    Бесплатная    9   
Юго-Восточный административный округ  Гагаринский район     Бесплатная    9   
Новомосковский административный округ поселение Сосенское      Платная    9   

                                                            max  mean  
Юго-Восточный административный округ  район Некрасовка       10  10.0  
Зеленоградский административный округ район Силино           10  10.0  
Северный административный округ       район Сокол            10  10.0  
Западный административный округ       район Марьино          10  10.0  
                                      район Филёвский Парк   10  10.0  
Юго-Восточный административный округ  район Печатники         9   9.0  
Западный административный округ       Можайский район         9   9.0  
                                      район Дорогомилово      9   9.0  
Юго-Восточный административный округ  Гагаринский район       9   9.0  
Новомосковский административный округ поселение Сосенское     9   9.0

In [77]:
df3_task.index.names = [None, None, 'type']

In [78]:
df3_task

min  \
                                                           type              
Юго-Восточный административный округ  район Некрасовка     Бесплатная   10   
Зеленоградский административный округ район Силино         Платная      10   
Северный административный округ       район Сокол          Платная      10   
Западный административный округ       район Марьино        Платная      10   
                                      район Филёвский Парк Бесплатная   10   
Юго-Восточный административный округ  район Печатники      Бесплатная    9   
Западный административный округ       Можайский район      Бесплатная    9   
                                      район Дорогомилово   Бесплатная    9   
Юго-Восточный административный округ  Гагаринский район    Бесплатная    9   
Новомосковский административный округ поселение Сосенское  Платная       9   

                                                                       max  \
                                                           type              
Юго-Восточный административный округ  район Некрасовка     Бесплатная   10   
Зеленоградский административный округ район Силино         Платная      10   
Северный административный округ       район Сокол          Платная      10   
Западный административный округ       район Марьино        Платная      10   
                                      район Филёвский Парк Бесплатная   10   
Юго-Восточный административный округ  район Печатники      Бесплатная    9   
Западный административный округ       Можайский район      Бесплатная    9   
                                      район Дорогомилово   Бесплатная    9   
Юго-Восточный административный округ  Гагаринский район    Бесплатная    9   
Новомосковский административный округ поселение Сосенское  Платная       9   

                                                                       mean  
                                                           type              
Юго-Восточный административный округ  район Некрасовка     Бесплатная  10.0  
Зеленоградский административный округ район Силино         Платная     10.0  
Северный административный округ       район Сокол          Платная     10.0  
Западный административный округ       район Марьино        Платная     10.0  
                                      район Филёвский Парк Бесплатная  10.0  
Юго-Восточный административный округ  район Печатники      Бесплатная   9.0  
Западный административный округ       Можайский район      Бесплатная   9.0  
                                      район Дорогомилово   Бесплатная   9.0  
Юго-Восточный административный округ  Гагаринский район    Бесплатная   9.0  
Новомосковский административный округ поселение Сосенское  Платная      9.0

### Сохраните и отправьте файл на степик.

In [79]:
df3_task.to_csv('df3_result.csv')

In [ ]:
#Решения однокурсников
df = pd.read_csv(archive.open('task5.csv'), index_col=[0, 1, 2], header=[0, 1])

res = df.rating.\
    sort_values('mean', ascending=False).\
    head(10).\
    reset_index(level=2).\
    rename(columns={'level_2': 'type'})  

## Задание 4 
### В одном из прошлых заданий мы выбирали все районы восточного административного округа и искали где больше всего мест на бесплатных парковках. Загрузите датафрейм из файла task5.csv и попробуйте найти в каком округе больше всего парковок (и платных и бесплатных вместе). Учитывайте только те районы, где средний рейтинг парковок больше или равен 5 

In [120]:
df4_task = pd.read_csv('task5.csv', index_col=[0, 1, 2], header=[0, 1]) # не редактируйте ячейку

In [97]:
df4_task

Capacity  \
                                                                           count   
Восточный административный округ Нижегородский район         Бесплатная        1   
                                 район Богородское           Бесплатная        2   
                                                             Платная           1   
                                 район Вешняки               Бесплатная        5   
                                                             Платная           3   
...                                                                          ...   
Южный административный округ     район Чертаново Северное    Платная           2   
                                 район Чертаново Центральное Бесплатная        7   
                                                             Платная           3   
                                 район Чертаново Южное       Бесплатная        9   
                                                             Платная           4   

                                                                             \
                                                                        sum   
Восточный административный округ Нижегородский район         Бесплатная   6   
                                 район Богородское           Бесплатная  14   
                                                             Платная      8   
                                 район Вешняки               Бесплатная  35   
                                                             Платная     17   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная     20   
                                 район Чертаново Центральное Бесплатная  68   
                                                             Платная     21   
                                 район Чертаново Южное       Бесплатная  86   
                                                             Платная     36   

                                                                        rating  \
                                                                           min   
Восточный административный округ Нижегородский район         Бесплатная      8   
                                 район Богородское           Бесплатная      8   
                                                             Платная         0   
                                 район Вешняки               Бесплатная      2   
                                                             Платная         0   
...                                                                        ...   
Южный административный округ     район Чертаново Северное    Платная         2   
                                 район Чертаново Центральное Бесплатная      0   
                                                             Платная         3   
                                 район Чертаново Южное       Бесплатная      3   
                                                             Платная         0   

                                                                             \
                                                                        max   
Восточный административный округ Нижегородский район         Бесплатная   8   
                                 район Богородское           Бесплатная  10   
                                                             Платная      0   
                                 район Вешняки               Бесплатная  10   
                                                             Платная      9   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная      6   
                                 район Чертаново Центральное Бесплатная   9   
                                                             Платная      9   
               

In [96]:
#Создаю маску для фильтрации всех парковок, у которых рейтинг >= 5
df4_task['rating']['mean'] >= 5

Восточный административный округ  Нижегородский район          Бесплатная     True
                                  район Богородское            Бесплатная     True
                                                               Платная       False
                                  район Вешняки                Бесплатная     True
                                                               Платная       False
                                                                             ...  
Южный административный округ      район Чертаново Северное     Платная       False
                                  район Чертаново Центральное  Бесплатная    False
                                                               Платная        True
                                  район Чертаново Южное        Бесплатная     True
                                                               Платная        True
Name: mean, Length: 247, dtype: bool

In [98]:
mask = df4_task['rating']['mean'] >= 5

In [106]:
df4_task = df4_task[mask]

In [119]:
df4_task.sum(level = 0)

C:\Users\user\AppData\Local\Temp\ipykernel_17008\3694237780.py:1: FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().
  df4_task.sum(level = 0)


Capacity      rating                 
                                           count  sum    min  max        mean
Восточный административный округ              35  216     61  119   92.450000
Западный административный округ               28  219     82  109   95.500000
Зеленоградский административный округ         13   94     19   36   28.100000
Новомосковский административный округ          8   96     20   35   27.500000
Северный административный округ               32  255     81  124  104.150000
Северо-Восточный административный округ      110  953     52  183  124.219170
Северо-Западный административный округ        14   99     18   40   28.866667
Центральный административный округ            98  305     20  100   63.394824
Юго-Восточный административный округ          35  248     66  115   91.000000
Юго-Западный административный округ           27  241     26   66   49.150000
Южный административный округ                  60  498     70  160  116.800000

In [123]:
#решения однокурсников
df4_task.loc[df4_task[('rating', 'mean')] >= 5, ('Capacity', 'count')].sum(level=0).idxmax()

C:\Users\user\AppData\Local\Temp\ipykernel_17008\724314144.py:1: FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().
  df4_task.loc[df4_task[('rating', 'mean')] >= 5, ('Capacity', 'count')].sum(level=0).idxmax()


'Северо-Восточный административный округ'

### Правильный ответ выберите на степике

## Задание 5
### Загрузите датафрейм из файла task5.csv. В этой задаче вы попробуете перейти от мультииндекса к обычному индексу с сохранением информации об уровнях. Сейчас на нулевом уровне колонки разбиваются на Capacity и rating, а на первом уровне Capacity и rating детализируются на count/sum и min/max/mean соответственно.
### Попытайтесь уйти от иерархии с сохранением информации о ней: сделайте обычный индекс с названием колонок:
#### Capacity_count, Capacity_sum, rating_min, rating_max, rating_mean

#### Не пытайтесь руками составить список и присвоить его т.к. на практике у вас может быть много колонок и руками перебирать будет долго. А еще можно что-то перепутать и тогда будет ошибка в последующих расчётах. Попробуйте перебрать значения мультииндекса в цикле и собрать тот самый список, который уже и присвоите обратно в датафрейм

In [124]:
df5_task = pd.read_csv('task5.csv', index_col=[0, 1, 2], header=[0, 1]) # не редактируйте ячейку

In [125]:
df5_task

Capacity  \
                                                                           count   
Восточный административный округ Нижегородский район         Бесплатная        1   
                                 район Богородское           Бесплатная        2   
                                                             Платная           1   
                                 район Вешняки               Бесплатная        5   
                                                             Платная           3   
...                                                                          ...   
Южный административный округ     район Чертаново Северное    Платная           2   
                                 район Чертаново Центральное Бесплатная        7   
                                                             Платная           3   
                                 район Чертаново Южное       Бесплатная        9   
                                                             Платная           4   

                                                                             \
                                                                        sum   
Восточный административный округ Нижегородский район         Бесплатная   6   
                                 район Богородское           Бесплатная  14   
                                                             Платная      8   
                                 район Вешняки               Бесплатная  35   
                                                             Платная     17   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная     20   
                                 район Чертаново Центральное Бесплатная  68   
                                                             Платная     21   
                                 район Чертаново Южное       Бесплатная  86   
                                                             Платная     36   

                                                                        rating  \
                                                                           min   
Восточный административный округ Нижегородский район         Бесплатная      8   
                                 район Богородское           Бесплатная      8   
                                                             Платная         0   
                                 район Вешняки               Бесплатная      2   
                                                             Платная         0   
...                                                                        ...   
Южный административный округ     район Чертаново Северное    Платная         2   
                                 район Чертаново Центральное Бесплатная      0   
                                                             Платная         3   
                                 район Чертаново Южное       Бесплатная      3   
                                                             Платная         0   

                                                                             \
                                                                        max   
Восточный административный округ Нижегородский район         Бесплатная   8   
                                 район Богородское           Бесплатная  10   
                                                             Платная      0   
                                 район Вешняки               Бесплатная  10   
                                                             Платная      9   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная      6   
                                 район Чертаново Центральное Бесплатная   9   
                                                             Платная      9   
               

In [151]:
df5_task.columns.values

array([('Capacity', 'count'), ('Capacity', 'sum'), ('rating', 'min'),
       ('rating', 'max'), ('rating', 'mean')], dtype=object)

In [160]:
df5_task.columns[1][1]

'sum'

In [161]:
df.columns = df.columns.str.replace('old_char', 'new_char')

TypeError: 'tuple' object does not support item assignment

### Сохраните и отправьте файл на степик.

In [ ]:
df5_result.to_csv('df5_result.csv')

## Задача 6
### Загрузите данные из task5.csv. По всей Москве рассчитайте:
### -Общее количество бесплатных парковок,
### -Общее количество платных парковок,
### -Общее количество мест на бесплатных парковках,
### -Общее количество мест на платных парковках

### Введите в поле с ответом получившиеся значения через пробел. В ответе должны быть целые числа
### Пример: 10 12 90 110

In [162]:
df6_task = pd.read_csv('task5.csv', index_col=[0, 1, 2], header=[0, 1]) 

In [163]:
df6_task

Capacity  \
                                                                           count   
Восточный административный округ Нижегородский район         Бесплатная        1   
                                 район Богородское           Бесплатная        2   
                                                             Платная           1   
                                 район Вешняки               Бесплатная        5   
                                                             Платная           3   
...                                                                          ...   
Южный административный округ     район Чертаново Северное    Платная           2   
                                 район Чертаново Центральное Бесплатная        7   
                                                             Платная           3   
                                 район Чертаново Южное       Бесплатная        9   
                                                             Платная           4   

                                                                             \
                                                                        sum   
Восточный административный округ Нижегородский район         Бесплатная   6   
                                 район Богородское           Бесплатная  14   
                                                             Платная      8   
                                 район Вешняки               Бесплатная  35   
                                                             Платная     17   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная     20   
                                 район Чертаново Центральное Бесплатная  68   
                                                             Платная     21   
                                 район Чертаново Южное       Бесплатная  86   
                                                             Платная     36   

                                                                        rating  \
                                                                           min   
Восточный административный округ Нижегородский район         Бесплатная      8   
                                 район Богородское           Бесплатная      8   
                                                             Платная         0   
                                 район Вешняки               Бесплатная      2   
                                                             Платная         0   
...                                                                        ...   
Южный административный округ     район Чертаново Северное    Платная         2   
                                 район Чертаново Центральное Бесплатная      0   
                                                             Платная         3   
                                 район Чертаново Южное       Бесплатная      3   
                                                             Платная         0   

                                                                             \
                                                                        max   
Восточный административный округ Нижегородский район         Бесплатная   8   
                                 район Богородское           Бесплатная  10   
                                                             Платная      0   
                                 район Вешняки               Бесплатная  10   
                                                             Платная      9   
...                                                                      ..   
Южный административный округ     район Чертаново Северное    Платная      6   
                                 район Чертаново Центральное Бесплатная   9   
                                                             Платная      9   
               

In [193]:
#Количество Бесплатных и Платных парковок
df6_task[('Capacity', 'count')].sum(level = 2)

C:\Users\user\AppData\Local\Temp\ipykernel_17008\1262265702.py:2: FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().
  df6_task[('Capacity', 'count')].sum(level = 2)


Бесплатная    518
Платная       529
Name: (Capacity, count), dtype: int64

In [195]:
df6_task[('Capacity', 'sum')].sum(level = 2)

C:\Users\user\AppData\Local\Temp\ipykernel_17008\1867552194.py:1: FutureWarning: Using the level keyword in DataFrame and Series aggregations is deprecated and will be removed in a future version. Use groupby instead. df.sum(level=1) should use df.groupby(level=1).sum().
  df6_task[('Capacity', 'sum')].sum(level = 2)


Бесплатная    3455
Платная       3642
Name: (Capacity, sum), dtype: int64

In [196]:
#Решения однокурсников
df = pd.read_csv('task5.csv', index_col=[0, 1, 2],  header=[0, 1])['Capacity']

df.groupby(level=2).sum()

count   sum
Бесплатная    518  3455
Платная       529  3642